core_backend/app/llm_call/parse_input.py

```diff
@@ -191,6 +194,68 @@ def _process_identified_language_response(
         return error_response
 
 
+def classify_on_off_topic__before(func: Callable) -> Callable:
+    """
+    Decorator to check if the question is on-topic or off-topic.
+    """
+
+    @wraps(func)
+    async def wrapper(
+        question: UserQueryRefined,
+        response: UserQueryResponse | UserQueryResponseError,
+        *args: Any,
+        **kwargs: Any,
+    ) -> UserQueryResponse | UserQueryResponseError:
+        """
+        Wrapper function to check if the question is on-topic or off-topic.
+        """
+        question, response = await _classify_on_off_topic(question, response)
+        response = await func(question, response, *args, **kwargs)
+        return response
+
+    return wrapper
+
+
+async def _classify_on_off_topic(
+    user_query: UserQueryRefined, response: UserQueryResponse | UserQueryResponseError
+) -> Tuple[UserQueryRefined, UserQueryResponse | UserQueryResponseError]:
+    """
+    Checks if the user query is on-topic or off-topic.
+    """
+    label = await _ask_llm_async(
+        question=user_query.query_text,
+        prompt=OnOffTopicClassification.get_prompt().format(
+            service_identity=SERVICE_IDENTITY
+        ),
+        litellm_model=LITELLM_MODEL_ON_OFF_TOPIC,
+    )
+
+    basic_cleaned = label.replace(" ", "_").upper()
+
+    try:
+        on_off_topic_label = getattr(OnOffTopicClassification, basic_cleaned)
```

Comment:
`getattr` takes a third argument that is the default value if it doesn't find the attribute. Might be able to reduce this all to one line

Reply:
@suzinyou this seems like a good idea, can you add or was there an issue?

Reply:
@suzinyou this seems like a good idea, can you add or was there an issue?